packages/react/src/primitives/Button/Button.tsx

```diff
@@ -24,14 +28,30 @@ const ButtonPrimitive: Primitive<ButtonProps, 'button'> = (
     size,
     type = 'button',
     variation,
+    colorTheme,
     ...rest
   },
   ref
 ) => {
+  // Test if variation supports color themes.
+  const supportsColorThemes = supportedVariations.includes(variation);
+
+  // Use the variation to construct our color theme modifier classes; in the
+  // case of our 'default' variation, use 'outlined'
+  const variationColorThemeModifier = variation ?? 'outlined';
```

Comment:
Should this depend on the resolved value of  `supportsColorThemes`?

Reply:
If I'm understanding your question, no, `variationColorThemeModifier` is used to build out our modifier class. In the case of our default button, where there is no variation defined, it uses the name 'outlined' to build out the class. In every other case, it uses the variation name. Then, on line 49~ish, we use `supportsColorTheme` (and if `colorTheme` exists) to determine if that modifier class should be applied. Does that make sense?

Reply:
Okay got it now. Since we don't reuse `variationColorThemeModifier` anywhere, maybe we just inline the logic on line 53 since it is only relevant if `supportsColorTheme` is truthy? Alternatively, think it is possible to abstract the relevant pieces here into a variable that resolves to either a `string` or `undefined` to pass to `classNameModifier`:
```
const colorThemeModifier = 
  supportedVariations.includes(variation) && colorTheme
  ? `${variation ?? 'outlined'}--${colorTheme}`
   : undefined;

const componentClasses = classNames(
  ComponentClassNames.Button,
  ComponentClassNames.FieldGroupControl,
  classNameModifier(ComponentClassNames.Button, variation),
  classNameModifier(ComponentClassNames.Button, colorThemeModifier),
  classNameModifier(ComponentClassNames.Button, size),
  // ...more modifiers
),
```

Reply:
I like that! Updated in the latest